crates/common/operation_pool/src/lib.rs

```diff
@@ -32,4 +43,53 @@ impl OperationPool {
                     >= beacon_state.finalized_checkpoint.epoch
             });
     }
+
+    // Proposer Slashing operations
+    pub fn has_slashing_for_proposer(&self, proposer_index: u64) -> bool {
+        self.proposer_slashings.read().contains_key(&proposer_index)
+    }
+
+    pub fn insert_proposer_slashing(
+        &self,
+        proposer_slashing: ProposerSlashing,
+    ) -> anyhow::Result<(), Error> {
+        let mut proposer_slashings = self.proposer_slashings.write();
+        let proposer_index = proposer_slashing.signed_header_1.message.proposer_index;
+
+        // Check if we already have a slashing for this validator
+        if proposer_slashings.contains_key(&proposer_index) {
+            return Err(Error::msg(
+                "Proposer slashing already exists for this validator",
+            ));
+        }
+
+        proposer_slashings
+            .entry(proposer_index)
+            .or_default()
+            .push(proposer_slashing);
+
+        let mut submission_time = self.slashing_submission_time.write();
+        submission_time.insert(proposer_index, Instant::now());
+
+        Ok(())
+    }
+
+    pub fn get_all_proposer_slashings(&self) -> Vec<ProposerSlashing> {
+        let proposer_slashings = self.proposer_slashings.read();
+        proposer_slashings
+            .values()
+            .flat_map(|v| v.iter())
+            .cloned()
+            .collect()
+    }
```

Comment:
```suggestion
    pub fn get_all_proposer_slashings(&self) -> Vec<ProposerSlashing> {
        self.proposer_slashings.read()
            .values()
            .flat_map(|v| v.iter())
            .cloned()
            .collect()
    }
```

might as well inline this